enhancements/ingress/set-delete-http-headers.md

```diff
@@ -1001,17 +1000,54 @@ This route can be annotated with `haproxy.router.openshift.io/disable_cookies=tr
+
+In case of HTTP response headers, the actions specified in `spec.httpHeaders.actions` on the IngressController will be executed after
+the actions specified in the Route's `spec.httpHeaders.actions` field.
+
+In the case of HTTP request headers, the actions specified in `spec.httpHeaders.actions[*].request` on the Route will be executed after
+the actions specified in the IngressController's `spec.httpHeaders.actions[*].request` field.
+
+ This is because of the way HAProxy is designed to work. We cannot change the processing order; we're constrained by HAProxy's design: 
+ the frontend receives the request and dispatches it to the backend, and then the frontend gets the 
+ response from the backend.
+
+This means that for requests, [the frontend's rules are applied first, followed by the backend's,
+and conversely](https://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-http-request), for responses, [the backend's rules are applied first, followed by the frontend's](https://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-http-response).
+ The `http-response set or delete` stanzas are applied and interpreted in this way by the HAProxy.
+For example: A cluster admin sets a response header with the name `X-Frame-Options` and the value `DENY` using the IngressController API. However, a route owner sets  the `X-Frame-Options` header with the value `SAMEORIGIN`.
+
+In this scenario, `haproxy.config` will look like the following where
+the response header called `X-Frame-Options` with value 'DENY' set in frontend will override the value `'SAMEORIGIN'` of `X-Frame-Options`
+which was set in the backend section via Route's spec.httpHeaders.actions.
+```conf
+frontend public
+  http-response set-header X-Frame-Options 'DENY' 
+
+frontend fe_sni
+  http-response set-header X-Frame-Options 'DENY' 
+
+frontend fe_no_sni
+  http-response set-header X-Frame-Options 'DENY' 
+
+backend be_secure:openshift-monitoring:alertmanager-main
+  http-response set-header X-Frame-Options 'SAMEORIGIN'
+```
+
+The following sequence diagram represents how the headers get set in sequence and how they could get over-ridden:
+```mermaid
+sequenceDiagram
+    client->>frontend: request (IngressController spec.httpHeaders.actions[].request[] applied)
+    frontend->>backend: request (Route spec.httpHeaders.actions[].request applied. If in presence of route annotations for the same header, spec.httpHeaders.actions[].request will be applied after it.)
+    backend->>application: request (application receives request headers)
+    application->>backend: response (application sets response headers)
+    backend->>frontend: response (Route spec.httpHeaders.actions[].response applied. If in presence of route annotations for the same header, spec.httpHeaders.actions[].request will be applied after it.)
+    frontend->>client: response (IngressController IngressController spec.httpHeaders.actions[].response applied)
+```
```

Comment:
Add the < b r > as mentioned so the diagram fits and can be read.  Also, I think you typed **request** where it should be **response** in one section.
```suggestion
```mermaid
sequenceDiagram
    client->>frontend: request - IngressController <br> spec.httpHeaders.actions[].request[] applied
    frontend->>backend: request - Route spec.httpHeaders.actions[].request applied. <br> If in presence of route annotations for the same header,<br>  spec.httpHeaders.actions[].request will be applied after it.
    backend->>application: request <br> (application receives request headers)
    application->>backend: response <br> (application sets response headers)
    backend->>frontend: response - Route spec.httpHeaders.actions[].response applied. <br> If in presence of route annotations for the same header, <br> spec.httpHeaders.actions[].response will be applied after it.
    frontend->>client: response - IngressController <br> IngressController spec.httpHeaders.actions[].response applied
```
```

Reply:
ok